lib/object_3d.dart

```diff
@@ -45,31 +57,66 @@ class _Object3DState extends State<Object3D> {
       // Load the object from a string
       _parseObj(widget.object!);
     }
+
+    final double maxSpeed = widget.maxSpeed.abs();
+    final double dampCoef = widget.dampCoef.clamp(0.001, 0.999);
+
+    _updateTimer = Timer.periodic(Duration(milliseconds: 16), (_) {
+      if (!mounted) return;
+      setState(() {
+        final adx = _deltaX.abs();
+        final ady = _deltaY.abs();
+        final sx = _deltaX < 0 ? -1 : 1;
+        final sy = _deltaY < 0 ? -1 : 1;
+
+        _deltaX = math.min(maxSpeed, adx) * sx * dampCoef;
+        _deltaY = math.min(maxSpeed, ady) * sy * dampCoef;
+
+        _yaw = _yaw - (_deltaX * (widget.reversePitch ? -1 : 1));
+        _pitch = _pitch - (_deltaY * (widget.reverseYaw ? -1 : 1));
+      });
+    });
+
     super.initState();
   }
 
+  @override
+  void dispose() {
+    super.dispose();
+    _updateTimer.cancel();
+  }
+
   /// Parse the object file.
   void _parseObj(String obj) {
     List<Vector3> vertices = [];
     List<List<int>> faces = [];
     final lines = obj.split("\n");
-    for (var line in lines) {
-      line = line.replaceAll(RegExp(r"\s+$"), "");
-      List<String> chars = line.split(" ");
-      if (chars[0] == "v") {
-        vertices.add(
-          Vector3(
-            double.parse(chars[1]),
-            double.parse(chars[2]),
-            double.parse(chars[3]),
-          ),
-        );
-      } else if (chars[0] == "f") {
-        List<int> face = [];
-        for (var i = 1; i < chars.length; i++) {
-          face.add(int.parse(chars[i].split("/")[0]));
+    for (String line in lines) {
+      const _space = " ";
```

Comment:
Although this repo does not have a linter set up, this is something the linter would complain about. Local variables do not have to be made private be prefixing them with `_` since they are already local.

Reply:
noted